plugin-infra/go-plugin-access/src/com/thoughtworks/go/plugin/access/elastic/ElasticAgentExtension.java

```diff
@@ -129,8 +129,13 @@ public ValidationResult onSuccess(String responseBody, String resolvedExtensionV
         });
     }
 
-    public String getStatusReport(String pluginId) {
+    public String getStatusReport(String pluginId, JobIdentifier identifier) {
```

Comment:
Probably I wouldn't modify the existing status report. 
I would look at having a separate view(job status report) when on the jobs details page which would give information specific to the job with information as status of the agent, errors if any while creating an agent etc.

Reply:
What does it even mean to say "get status report for a job"? I'd think a status report should show the status of all nodes in the cluster along with the pods/services and what job they map to.

I believe one of the problems is that a container may be started for a particular job, but may possibly be assigned to another job because `should-assign-work` returns true. I don't believe there's any way to know what job an elastic agent is running, unless the server **explicitly** tells the plugin about it.

You could certainly make **assumptions** about it based on the job identifiers passed to the  `create-agent` and `should-assign-work` — but they'll be just that — assumptions, and they'll be wrong in some cases...

Reply:
Another option is to do this in JS, as the job ID will be in the URL. The page (either through JS or using the passed in identifier) can decide whether to highlight the job/pod/service by hiding everything else (a job status report) or keeping everything else intact and highlighting the job defined by the passed in ID.

Reply:
It's true that the server is the only one that knows which GoCD agent a job is running on. Isn't it also the one that knows which elastic agent ID that GoCD agent has? If so, instead of depending on job ID, shouldn't it be sending elastic agent ID and tell the status report to highlight that?

Reply:
Maybe worth talking about it during a call.

Reply:
Writing down what I am thinking just that I don't forget.

When I am on the Job details page I would prefer to see details of an agent in context of the specific job rather than seeing an entire Status report. If the user finds the agent information insufficient the user should be given a option to view the entire status report. 
Anyways lets talk later.